sql/api/src/main/scala/org/apache/spark/sql/types/StringType.scala

```diff
@@ -39,6 +39,9 @@ class StringType private(val collationId: Int) extends AtomicType with Serializa
   def supportsBinaryEquality: Boolean =
     CollationFactory.fetchCollation(collationId).supportsBinaryEquality
 
+  def supportsLowercaseEquality: Boolean =
+    CollationFactory.fetchCollation(collationId).supportsLowercaseEquality
+
   def isUTF8BinaryCollation: Boolean =
```

Comment:
Do we expect end users to call these functions? If not we should mark them as `private[sql]`